benchmarks/functional_autograd_benchmark/README.md

```diff
@@ -0,0 +1,16 @@
+# Benchmarking tool for the autograd API
+
+This folder contain a set of self-contained scripts that allow to benchmark the autograd with different common models.
+It is designed to run the benchmark before and after your change and will generate a table to share on the PR.
+
+To do so, you can use `functional_autograd_benchmark.py` to run the benchmarks before your change (using as output `before.txt`) and after your change (using as output `after.txt`).
+You can then use `compare.py` to get a markdown table comparing the two runs.
+
+The default arguments of `functional_autograd_benchmark.py` should be used in general. You can change them though to force a given device or force running even the (very) slow settings.
+
+
+### Files in this folder:
+- `functional_autograd_benchmark.py` is the main entry point to run the benchmark.
+- `compare.py` is the entry point to run the comparison script that generates a markdown table.
+- `torchaudio_models.py` and `torchvision_models.py`  contains code extracted from torchaudio and torchvision to be able to run the models without having a specific version of these libraries installed.
+- `ppl_models.py`, `vision_models.py` and `audio_text_models.py` contain all the getter functions used for the benchmark.
```

Comment:
Should we add a test for the benchmark suite? From past experience (I co-wrote `benchmarks/fastrnns/bench.py`), benchmarks can and will bit-rot pretty fast (e.g., it runs in one version of PyTorch and stops running in the next). Something like a smoke test that just tries to run a few benchmarks and checks that nothing blows up would be nice to have.

Reply:
Do you have pointers on where to add these smoke tests?

Reply:
One approach could be:
- add a `test/test_autograd_functional_benchmark.py` file
- Inside `test/test_autograd_functional_benchmark.py`, have a test that does `subprocess.run(['python', '../benchmarks/functional_autograd_benchmark.py', '--output', '<some temporary file>']) and asserts that the output file is non-empty and the return code was 0.

Another approach could be to refactor the benchmarks folder so that it is importable from `test/test_autograd_benchmark.py`. That way the testing code doesn't have to use `subprocess.run`. But that (making the benchmarks folder importable from `test/test_autograd_benchmark.py`) sounds pretty painful.